### ISSUE TITLE
Unnecessary `this` tempvar-ing

### ISSUE THREAD

**Issue by RealyUniqueName** [HaxeFoundation/haxe#9773]
```haxe
class Main {
	static public function main() {}

	function doSomething() {
		use(() -> test());
	}

	@:pure(false) function test() {}

	@:pure(false) static function use(fn:()->Void) {}
}
```
```
$ haxe -main Main -D analyzer-optimize -D dump=pretty
```
Dump for `doSomething`:
```
	function doSomething() {
		var _gthis = this;
		Main.use(function() {
			_gthis.test();
		});
	}
```

**Comment by Simn** [comment]
Isn't that necessary?

**Comment by RealyUniqueName** [comment]
Not for all targets.

**Comment by Simn** [comment]
Then we need a platform config for that.

Note that not all generators might deal with this correctly because it introduces a new case with the captured `this`. We can try this on a branch and see what happens, and then look into fixing potential issues as a second step.

**Comment by nadako** [comment]
The original example is necessary for javascript, or am I confusing things?

**Comment by Simn** [comment]
Yes, we're trying to determine on which targets it's not necessary.

... and according to my blunt test, the only target where this just works is python of all things. x)

**Comment by RealyUniqueName** [comment]
And php )
Using `this` in closures 100% works in PHP. The failure of Simn's commit is probably caused by something else.
